### ISSUE TITLE
[msschem] Not all hsecs are being offered by the client

### ISSUE THREAD

**Issue by ReimarBauer** [Open-MSS/MSS#163]
**[Original report](https://bitbucket.org/wxmetvis/mss/issue/163) by Andreas Hilboll (Bitbucket: [andreas-h](https://bitbucket.org/andreas-h), GitHub: [andreas-h](https://github.com/andreas-h)).**

----------------------------------------

**[This replaces https://bitbucket.org/andreas-h/mss/issues/1]**

In my setup, I add the following two data sources to the `nwpaccess` in the settings:

    "CAMSglb": mslib.mswms.dataaccess.MSSChemDataAccess(
            os.path.join(_datapath, "msschem", "cams_global"),
            "CAMSglb", "cams-global", "ml"),

    "CAMSregENS": mslib.mswms.dataaccess.MSSChemDataAccess(
            os.path.join(_datapath, "msschem", "cams_regional"),
            "CAMSregENS", "cams-regional", "al"),

I also have the following two horizontal plot styles:

    (mpl_hsec_styles.HS_MSSChemStyle_AL_NO2_mconc, ["CAMSregENS"]),
    (mpl_hsec_styles.HS_MSSChemStyle_ML_NO2_mfrac, ["CAMSglb"]),

Everything works fine, as long as I **don't** add the following vertical plot style:

    (mpl_vsec_styles.VS_MSSChemStyle_ML_NO2_mfrac, ["CAMSglb"]),

As soon as I also have this vertical layer activated, the client only offers the CAMSregENS hsec, and *not* the CAMSglb hsec.

However, I checked the WMS offer from the server (using the *View* functionality of the client), and the offers in both situations are idential (ok, except for the added vsec style).

I'm a bit at a loss here.  Since the WMS offer from the server seems identical, I don't know where to look in my server code, and I haven't touched the client, so far ...

I uploaded some test data (2.5G) to [OneDrive](https://unibremende-my.sharepoint.com/personal/hilboll_uni-bremen_de/_layouts/15/guestaccess.aspx?folderid=1a66e6db13b17483f98c63af4bac53b6a&authkey=AfmhPn50oyWMiQfG5Jv9uQ8).

**Comment by ReimarBauer** [comment]
**Original comment by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

How does the full mss_wms_settings file look like?

**Comment by ReimarBauer** [comment]
**Original comment by Andreas Hilboll (Bitbucket: [andreas-h](https://bitbucket.org/andreas-h), GitHub: [andreas-h](https://github.com/andreas-h)).**

----------------------------------------

**Comment by ReimarBauer** [comment]
**Original comment by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

This should help:


```
#!python

--- a/mslib/mswms/mpl_vsec_styles.py
+++ b/mslib/mswms/mpl_vsec_styles.py
@@ -1544,7 +1544,7 @@ def make_msschem_class(entity, nam, vert, units, scale, add_data=None, add_conto
         add_contours = []

     class fnord(VS_MSSChemStyle):
-        name = entity + "_" + vert
+        name = "VS_" + entity + "_" + vert
         dataname = entity
         ###units, unit_scale = Targets.get_unit(dataname)
         units = units

```

Each offered layer need to have a unique "name" otherwise the client cannot distinguish them.

**Comment by ReimarBauer** [comment]
**Original comment by Andreas Hilboll (Bitbucket: [andreas-h](https://bitbucket.org/andreas-h), GitHub: [andreas-h](https://github.com/andreas-h)).**

----------------------------------------

thanks, @joernu76 , see pull request #297

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

BUG: fix duplicate layer name problem

fixes #163
